packages/js/product-editor/src/blocks/product-fields/shipping-class/edit.tsx

```diff
@@ -122,15 +122,22 @@ export function Edit( {
 		throw error;
 	}
 
-	const { shippingClasses } = useSelect( ( select ) => {
-		const { getProductShippingClasses } = select(
-			EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME
-		);
-		return {
-			shippingClasses:
-				getProductShippingClasses< ProductShippingClass[] >() ?? [],
-		};
-	}, [] );
+	const { shippingClasses } = useSelect(
+		( select ) => {
+			const { getProductShippingClasses } = select(
+				EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME
+			);
+			let classes: ProductShippingClass[] = [];
+			if ( isInSelectedTab ) {
+				classes =
+					getProductShippingClasses< ProductShippingClass[] >() ?? [];
+			}
+			return {
+				shippingClasses: classes,
+			};
```

Comment:
could you try to use the tertiary structure here?

```suggestion
			return isInSelectedTab
				? getProductShippingClasses< ProductShippingClass[] >() ?? []
				: [];
```

I haven't tested, just in case 😅

Reply:
Thank you @retrofox for the suggestion.
I think that change won't cover what Maikel [pointed out here](https://github.com/woocommerce/woocommerce/pull/47196#discussion_r1592413291).
But there is some room for improvement in my code. I changed it in the commit [b1a5710](https://github.com/woocommerce/woocommerce/pull/47196/commits/b1a571057ef2f2ccf4bd6306571cee352845917d).

Reply:
> I think that change won't cover what Maikel https://github.com/woocommerce/woocommerce/pull/47196#discussion_r1592413291.

I'm not sure you need to be concerned about `shippingClasses` being `null`. [Here](https://github.com/woocommerce/woocommerce/blob/7a7b5137716623a7b8d13658d0a1a71228db9b0a/packages/js/product-editor/src/blocks/product-fields/shipping-class/edit.tsx#L161) is already covered.
And probably [here](https://github.com/woocommerce/woocommerce/blob/7a7b5137716623a7b8d13658d0a1a71228db9b0a/packages/js/product-editor/src/blocks/product-fields/shipping-class/edit.tsx#L63) too, when calling conditionally to the `some` method. You could check the type there, too.
Anyway, if you want to ensure the type of the being array, you could do something like the following:

```jsx
const shippingClasses =
  useSelect(
    ( select ) => {
      return isInSelectedTab
        ? select(
            EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME
          ).getProductShippingClasses< ProductShippingClass[] >()
        : [];
    },
    [ isInSelectedTab ]
  ) || [];
```

it's up to you :-)